From 5abee7da185f4887d1596e216acce03929a42d1a Mon Sep 17 00:00:00 2001
From: hht8 <huanghaitao12138@163.com>
Date: Fri, 20 Aug 2021 15:11:08 +0800
Subject: [PATCH] 2

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index fc29d92..c36dc47 100644
--- a/Makefile
+++ b/Makefile
@@ -26,7 +26,7 @@ path_to_add := $(addsuffix /bin,$(subst :,/bin:,$(GOPATH))):$(PWD)/tools/bin
 export PATH := $(path_to_add):$(PATH)
 
 GO              := GO111MODULE=on go
-GOBUILD         := $(GO) build $(BUILD_FLAG) -tags codes
+GOBUILD         := $(GO) build $(BUILD_FLAG) --mod=vendor -tags codes
 GOBUILDCOVERAGE := GOPATH=$(GOPATH) cd tidb-server; $(GO) test -coverpkg="../..." -c .
 GOTEST          := $(GO) test -p $(P)
 OVERALLS        := GO111MODULE=on overalls
-- 
2.30.0

